### DISCUSSION TITLE
How to fetch kv cache under llamacpp like “past_key_values“？

### DISCUSSION THREAD

**Discussion by hitdra** [ggml-org/llama.cpp#9670] [Q&A]
With pytorch and the transformer framework, I can get the kv cache like this
`generated = model.generate(input_ids, max_new_tokens = 1, return_dict_in_generate=True)`
`kv = generated['past_key_values']`
How to get the corresponding kv cache under llamacpp?
I really appreciate everyone being able to answer my questions, thanks a million!

**Comment by bingo787** [comment]
I want to konw also.
